projects/uswds-components/src/lib/input/affix.directive.ts

```diff
@@ -0,0 +1,79 @@
+import { AfterViewInit, Directive, ElementRef, Input, Renderer2, TemplateRef, ViewContainerRef } from '@angular/core';
+
+@Directive({
+  selector: 'input[prefix], input[suffix]'
+})
+export class UsaInputAffixDirective implements AfterViewInit {
+
+  @Input() prefix: TemplateRef<any> | string;
+  @Input() suffix: TemplateRef<any> | string;
+
+  constructor(
+    private el: ElementRef,
+    private renderer: Renderer2,
+    private viewContainer: ViewContainerRef) {
+  }
+
+  ngAfterViewInit(){
+
+    const parent = this.renderer.parentNode(this.el.nativeElement);
+
+    //Need next sibling to ensure generated usa-input-group is inserted into the same place on the page and input is expected to be
+    const nextSibling = this.renderer.nextSibling(this.el.nativeElement);
+
+    let prefixElement, suffixElement;
+    if(this.prefix){
+      prefixElement = this.handleAffix(this.prefix, 'usa-input-prefix')
+    }
+    if(this.suffix){
+      suffixElement = this.handleAffix(this.suffix, 'usa-input-suffix')
+    }
+
+    const inputGroupDiv = document.createElement('div');
+    this.renderer.addClass(inputGroupDiv,'usa-input-group')
+
+    // Apply input class in case only directive was applied
+    this.renderer.addClass(this.el.nativeElement, 'usa-input')
+
+    if(this.prefix){
+      this.renderer.appendChild(inputGroupDiv, prefixElement);
+    }
+    this.renderer.appendChild(inputGroupDiv, this.el.nativeElement);
+    if(this.suffix){
+      this.renderer.appendChild(inputGroupDiv, suffixElement);
+    }
```

Comment:
Here, I see that the logic in this if statement and the if statement on like 38 do the same thing. Instead of creating a `prefixElement` and a `suffixElement`, can we just create an `affixElement` in line 24, set that as needed in line 26 / 29, and then here remove the if statements?
```
let affix;
if (this.prefix) {
   affix = ...
 } else {
   affix = ...
 }

 // and then here - 
 this.renderer.appendChild(inputGroupDiv, affix);
```
This way we can remove the additional if checks on line 38 / 42

Reply:
Where the appendChild is relative to this.el.nativeElement is important, prefixes must be before, suffixes must be after. So logic is still needed to place the affix in the right place.

Also, using a single var, affixElement, would work if only prefix or suffix were being used, but will not correctly if both are being used.

Reply:
Ah I see what you're saying. Yes, the relativity makes sense